include/modem/location.h

```diff
@@ -185,11 +179,28 @@ struct location_data_error {
 };
 #endif
 
+/** Cloud location information. */
+struct location_data_cloud {
+#if defined(CONFIG_LOCATION_SERVICE_EXTERNAL) && defined(CONFIG_LOCATION_METHOD_CELLULAR)
+	/** Cellular cell information. */
+	const struct lte_lc_cells_info *cell_data;
+#endif
+#if defined(CONFIG_LOCATION_SERVICE_EXTERNAL) && defined(CONFIG_LOCATION_METHOD_WIFI)
+	/** Wi-Fi access point information. */
+	const struct wifi_scan_info *wifi_data;
+#endif
+};
+
 /** Location event data. */
 struct location_event_data {
 	/** Event ID. */
 	enum location_event_id id;
-	/** Used location method. */
+	/**
+	 * Used location method.
+	 *
+	 * When cellular and Wi-Fi positioning are used and they are combined into a single
+	 * cloud request by the library, there is some uncertainty on the reported location method.
```

Comment:
Can you be more specific here instead of just saying that there's some uncertainty? What's the logic here?

Reply:
It's WiFi if we don't have more information like in timeouts and error cases as well as `LOCATION_EVT_CLOUD_LOCATION_EXT_REQUEST` and `LOCATION_EVT_RESULT_UNKNOWN` if WiFi scan returned more than one AP.
When location is found, if one didn't return scan results, it's the other one. If both returned scan results and accuracy is available, returned method is cellular if accuracy is higher than `METHOD_WIFI_ACCURACY_THRESHOLD`, otherwise Wi-Fi.

So, we are making an educated guess and actually in timeouts and errors we are kind of lying because it's not WiFi necessarily in those cases. Do you want me to write all details or what?

Reply:
Tried something. Please check.